component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/util/Clusterer.java

```diff
@@ -332,9 +334,15 @@ private Cluster findAssociatedCluster(DataPoint currentDatapoint) {
             double[] total;
             total = new double[dimensionality];
             for (DataPoint d: c.getDataPointsInCluster()) {
-                Arrays.setAll(total, i -> total[i] + d.getCoordinates()[i]);
+                double[] coordinatesOfd = d.getCoordinates();
+                for (int i = 0; i < dimensionality; i++) {
+                    total[i] += coordinatesOfd[i];
+                }
+            }
+            int numberOfMembers = c.getDataPointsInCluster().size();
```

Comment:
try not to use single character for variable names. Provide meaningful ones

Reply:
Fixed in https://github.com/siddhi-io/siddhi-execution-streamingml/pull/76